### PR TITLE
Fixes #20080 - Allow runners to set timeout

### PR DESCRIPTION
**Pull request by adamruzicka**
- [x] add info about timeout to continuous output (https://github.com/theforeman/foreman-tasks/pull/258/files#r123245188)
- [X] add separate issue for this change, for better tracking of what needs to be done where and why

### CODE REVIEW

File: lib/foreman_tasks_core/runner/base.rb [hunk 1]
```diff
@@ -41,6 +41,16 @@ def close
         # if cleanup is needed
       end
 
+      def timeout
+        # Override when timeouts and regular kills should be handled differently
+        kill
```
Comment (iNecas) [inline lib/foreman_tasks_core/runner/base.rb]: What about adding a debug message to `@continuous_output` about the timeout passed, something like `Time out for the execution passed, trying to kill` (perhaps the working could be better: it sounds pretty violent when put out of the context :)

**Comment by iNecas** [comment]
I will agree with the prprocessor here: please add a separate issue for this so that we can track the differences needed + assign the proper tasks version to it once merged.

**Comment by adamruzicka** [comment]
Comments addressed

**Comment by iNecas** [comment]
Thanks @adamruzicka

**Pull request merged**